src/blockedpermutation.jl

```diff
@@ -70,12 +70,17 @@ function blockedperm(permblocks::Union{Tuple{Vararg{Int}},Int}...; kwargs...)
   return blockedperm(collect_tuple.(permblocks)...; kwargs...)
 end
 
-function blockedperm(permblocks::Union{Tuple{Vararg{Int}},Int,Ellipsis}...; kwargs...)
+function blockedperm(
+  permblocks::Union{Tuple{Vararg{Int}},Tuple{Ellipsis},Int,Ellipsis}...; kwargs...
+)
   return blockedperm(collect_tuple.(permblocks)...; kwargs...)
 end
 
-function blockedperm(bt::AbstractBlockTuple)
-  return blockedperm(Val(length(bt)), blocks(bt)...)
+# keep len kwarg to be consistent with other method signatures
+function blockedperm(bt::AbstractBlockTuple; length::Union{Val,Nothing}=nothing)
+  !(length ∈ (nothing, Val(Base.length(bt)))) &&
+    throw(ArgumentError("Invalid total length"))
+  return blockedperm(Val(Base.length(bt)), blocks(bt)...)
```

Comment:
This is mostly aesthetic, but I think a nicer way to handle this logic would be:
```julia
function blockedperm(bt::AbstractBlockTuple; length=nothing)
  return _blockedperm(length, bt)
end

function _blockedperm(::Nothing, bt::AbstractBlockTuple)
  return _blockedperm(Val(length(bt)), bt)
end

function _blockedperm(vallength::Val, bt::AbstractBlockTuple)
  vallength == Val(length(bt)) || throw(ArgumentError("Invalid total length"))
  # ...
end
```